doc/reference/actor.md

```diff
@@ -2,755 +2,738 @@
-::: tip usage
+### default-known-servers
+```scheme
+(default-known-servers)
 ```
-(start-rpc-server! ...)
-```
-:::
 
-Please document me!
+Returns a hash table with the default known servers.
+By default it only contains the registry reachable in its default address.
 
-### stop-rpc-server!
-::: tip usage
-```
-(stop-rpc-server! ...)
+### set-default-known-servers!
+```scheme
+(set-default-known-servers! servers)
 ```
-:::
 
-Please document me!
+Sets the default known servers; `servers` must be a hash table mapping
+server identifiers to lists of addresses.
 
-### rpc-connect
-::: tip usage
-```
-(rpc-connect ...)
+### default-registry-addresses
+```scheme
+(default-registry-addresses)
 ```
-:::
 
-Please document me!
+Returns the default registry addresses, as a list of addresses.  By
+default the registry is reachable at `/tmp/ensemble/registry` in the
+current host.
 
-### rpc-register
-::: tip usage
-```
-(rpc-register ...)
+### set-default-registry-addresses!
+```scheme
+(set-default-registry-addresses! addrs)
 ```
-:::
 
-Please document me!
+Sets the default registry addresses.
 
-### rpc-unregister
-::: tip usage
+### server-address-cache-ttl
+```scheme
+(server-address-cache-ttl)
 ```
-(rpc-unregister ...)
-```
-:::
 
-Please document me!
+Returns the actor's server address cache TTL in seconds; by default this is 5 minutes.
 
-### bind-protocol!
-::: tip usage
-```
-(bind-protocol! ...)
+### set-server-address-cache-ttl!
+```scheme
+(set-server-address-cache-ttl! ttl)
 ```
-:::
 
-Please document me!
+Sets the actor server's address cache TTL (in seconds).
 
 
-## RPC Server Protocol
 
-### rpc.register
-::: tip usage
-```
-(rpc.register ...)
-(!rpc.register ...)
-(!!rpc.register ...)
+## Ensemble Servers
+
+### call-with-ensemble-server
+```scheme
+(call-with-ensemble-server server-id thunk
+                           log-level: (log-level 'INFO)
+                           log-file:  (log-file #f)
+                           listen:    (listen-addrs [])
+                           announce:  (public-addrs #f)
+                           registry:  (registry-addrs #f)
+                           roles:     (roles [])
+                           cookie:    (cookie (get-actor-server-cookie)))
 ```
-:::
 
-Please document me!
+This is the programmatic equivalent of `gxensemble run`; first it
+starts the logger with the appropriate options, and then it starts a
+new actor server with identifier `server-id`, starts the loader
+service, adds the server to the ensemble, and then invokes thunk.
+When the thunk exits, it shuts down the actor server and removes the
+server-id from the ensemble.
 
-### rpc.unregister
-::: tip usage
-```
-(rpc.unregister ...)
-(!rpc.unregister ...)
-(!!rpc.unregister ...)
+Options:
+- `log-level`: logging level to use; INFO by default
+- `log-file`: log file for the logger; if it is "-" then the canonical
+  server log is used; this file is at
+  `$GERBIL_PATH/ensemble/server/<server-id>/log`.
```

Comment:
What format for the log file?

Shouldn't we follow the XDG spec here and/or some systematic derivation thereof?
See `:clan/path-config` and `:clan/config`
e.g. ${XDG_DATA_HOME}/ensemble/log/server/<server-id> ?
or ${XDG_DATA_HOME}/gerbil/log/ensemble/server/<server-id> ?

Reply:
No, I don't want this XDG crap anywhere near our code.

The log format is the format of the logger users, which is pretty standard when it comes to loggers.
Out of scope, we don't need to specify it here -- we have a standard logger package after all.